.github/workflows/continuous-integration.yml

```diff
@@ -0,0 +1,171 @@
+# Configuration for testing and deploying with GitHub Actions
+name: CI
+
+# Only build PRs, the master branch, and tags. Pushes to branches will only be
+# built when a PR is opened.
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+    tags:
+      - v*
+
+# Use bash by default in all jobs
+defaults:
+  run:
+    # The -l {0} is necessary for conda environments to be activated
+    shell: bash -l {0}
+
+###############################################################################
+jobs:
+
+  # Check code style and linting. Run separately so this finishes fast and
+  # reports problems right away.
+  style_checks:
```

Comment:
You may want to consider a workflow like [this one](https://github.com/matplotlib/matplotlib/blob/master/.github/workflows/reviewdog.yml) that matplotlib uses for style checks. You end up with comments inline with the source. I have not yet set it up myself, but it's on the short list.

Reply:
That would be useful for the flake8 run for sure! I'll have a look at that. It would be better to the have the style checks as a separate workflow. It kind of sucks that we can't rerun failing builds only on Actions.